### DISCUSSION TITLE
Refetching with a different chain

### DISCUSSION THREAD

**Discussion by sunnyvempati** [wevm/wagmi#561] [Q&A]
Hello!

When switching network using `switchNetwork`, it seems that the `useContractRead` hooks on the page don't refetch the data with the new network in mind automatically.  I also cannot call `refetch` because it still uses the old `chainId`.   

As a note: if i switch wallets within metamask, the page does refetch the data however.

My question is:

How would I refetch data on the page when user switches network using the custom connect button?

**Comment by tmm** [comment]
> When switching network using switchNetwork, it seems that the useContractRead hooks on the page don't refetch the data with the new network in mind automatically.

They should if your provider is [chain-aware](https://wagmi.sh/docs/client#provider-optional=). Can take a look if you add a basic CodeSandbox.

> How would I refetch data on the page when user switches network using the custom connect button?

Have you tried asking this in [RainbowKit's Discussions](https://github.com/rainbow-me/rainbowkit/discussions)?
